src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -130,6 +130,25 @@ begin
   exact h.closure_eq
 end
 
+/-- The infimum edistance of a point to a set is positive if and only if the point is not in the
+closure of the set. -/
+lemma inf_edist_pos_iff_not_mem_closure {x : α} {E : set α} :
+  0 < inf_edist x (closure E) ↔ x ∉ closure E :=
+by simp only [mem_iff_inf_edist_zero_of_closed is_closed_closure, zero_lt_iff]
+
+lemma exists_real_pos_le_infdist_of_not_mem_closure {x : α} {E : set α} (h : x ∉ closure E) :
+  ∃ (ε : ℝ), 0 < ε ∧ ennreal.of_real ε ≤ inf_edist x E :=
```

Comment:
You can use https://leanprover-community.github.io/mathlib_docs/data/real/ennreal.html#ennreal.lt_iff_exists_nnreal_btwn or one of the nearby lemmas instead of this one.